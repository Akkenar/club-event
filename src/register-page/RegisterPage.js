import React, { useContext, useEffect } from 'react';
import RegisterFormContainer from './register-form/RegisterFormContainer';
import { Header } from 'semantic-ui-react';
import getKey from '../core/intl/getKey';
import { goToTop, setPageTitle } from '../page.lib';
import LanguageContext from '../core/intl/LanguageContext';

const RegisterPage = () => {
  const { messages } = useContext(LanguageContext);

  useEffect(() => {
    setPageTitle(getKey('register.page.title', messages));
    goToTop();
  });

  return (
    <div>
      <Header as="h1">{getKey('register.page.title', messages)}</Header>
      <RegisterFormContainer />
    </div>
  );
};

export default RegisterPage;
