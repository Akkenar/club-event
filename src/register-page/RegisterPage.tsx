import React from 'react';
import { useContext, useEffect } from 'react';
import { Header } from 'semantic-ui-react';
import getKey from '../core/intl/getKey';
import LanguageContext from '../core/intl/LanguageContext';
import { goToTop, setPageTitle } from '../core/page.lib';
import RegisterFormContainer from './register-form/RegisterFormContainer';

const RegisterPage = () => {
  const { messages } = useContext(LanguageContext);

  useEffect(() => {
    setPageTitle(getKey('register.page.title', messages));
    goToTop();
  });

  return (
    <div>
      <Header as="h1" data-testid="main-title">
        {getKey('register.page.title', messages)}
      </Header>
      <RegisterFormContainer />
    </div>
  );
};

export default RegisterPage;
