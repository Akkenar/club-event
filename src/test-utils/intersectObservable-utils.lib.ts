import { mockGlobalProperty } from './test-utils.lib';

const allObservers: IntersectionObserverMock[] = [];

class IntersectionObserverMock {
  private readonly cb: any;
  public constructor(cb: any) {
    this.cb = cb;

    // To allow global triggering.
    allObservers.push(this);
  }

  public trigger() {
    this.cb([{ isIntersecting: true }]);
  }

  public observe() {
    // Nothing
  }

  public unobserve() {
    // Nothing.
  }

  public disconnect() {
    // Nothing.
  }
}

class IntersectionObserverEntryMock {
  public intersectionRatio() {
    // Nothing.
  }

  public isIntersecting() {
    // Nothing.
  }
}

export function mockInterestObservable(): () => void {
  // Reset all the mocks
  allObservers.slice(0);

  mockGlobalProperty(window)('IntersectionObserver')(IntersectionObserverMock);
  mockGlobalProperty(window)('IntersectionObserverEntry')(
    IntersectionObserverEntryMock,
  );

  // Trigger all.
  return () => {
    allObservers.forEach((observer: IntersectionObserverMock) => {
      observer.trigger();
    });
  };
}
