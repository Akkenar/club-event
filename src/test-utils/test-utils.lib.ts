import { fireEvent } from '@testing-library/react';

export function mockGrecaptcha() {
  (window as any).grecaptcha = {
    getResponse: () => 'response',
    render: jest.fn(),
    reset: jest.fn(),
  };
}

export function mockResponse(response: any) {
  const mock = typeof response === 'string' ? { result: response } : response;

  // Mock the fetch data as JSDom doesn't implement fetch.
  // @ts-ignore
  window.fetch = jest.fn(() =>
    Promise.resolve({
      text: () =>
        new Promise(resolve => {
          const mockResult = JSON.stringify(mock);
          setTimeout(() => {
            resolve(mockResult);
          }, 50);
        }),
    }),
  );
}

export function setValidInputData(
  validData: any,
  { getByTestId }: { getByTestId: any },
) {
  Object.keys(validData).forEach((key: string) => {
    fireEvent.change(getByTestId(key), {
      // @ts-ignore
      target: { value: validData[key] },
    });
  });
}

export const mockGlobalProperty = (globalObject: any) => (key: string) => (
  value: any,
) => {
  const original = globalObject[key];
  Object.defineProperty(globalObject, key, { value, writable: true });
  return () => mockGlobalProperty(globalObject)(key)(original);
};

export function forceLoadImage() {
  Object.defineProperty(Image.prototype, 'onload', {
    configurable: true,
    set(fn: () => void) {
      // Call the function directly to simulate the loading
      fn();
    },
  });
}
