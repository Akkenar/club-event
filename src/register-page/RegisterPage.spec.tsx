import * as React from 'react';
import { render } from '@testing-library/react';
import { mockInterestObservable } from '../test-utils/intersectObservable-utils.lib';
import { mockGrecaptcha } from '../test-utils/test-utils.lib';
import RegisterPage from './RegisterPage';

mockGrecaptcha();

describe('RegisterPage', () => {
  beforeEach(() => {
    mockInterestObservable();
  });

  it('should match the snapshot', () => {
    const wrapper = render(<RegisterPage />);
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should set the page title', () => {
    // Will be set by side effect through a hook
    render(<RegisterPage />);
    expect(document.title).toEqual('register.page.title');
  });
});
