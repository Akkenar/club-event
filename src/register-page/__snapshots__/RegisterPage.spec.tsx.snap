// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPage should match the snapshot 1`] = `
<body>
  <div>
    <div>
      <h1
        class="ui header"
        data-testid="main-title"
      >
        register.page.title
      </h1>
      <form
        class="ui form RegisterForm"
        novalidate=""
      >
        <div
          class="ui visible warning message"
        >
          <div
            class="content"
          >
            <div>
              register.warning
            </div>
          </div>
        </div>
        <div
          class="ui error message"
          data-testid="backend-error"
        >
          <div
            class="content"
          >
            <div
              class="header"
            >
              register.form.error.title
            </div>
            <p>
              register.form.error.message
            </p>
          </div>
        </div>
        <p>
          register.form.description
        </p>
        <div
          class="ui segment"
        >
          <h2
            class="ui dividing header"
          >
            register.form.personalInfo
          </h2>
          <div
            class="equal width fields"
          >
            <div
              class="required field"
            >
              <label
                for="firstName"
              >
                register.form.firstName
              </label>
              <input
                data-testid="firstName"
                id="firstName"
                maxlength="255"
                name="firstName"
                placeholder="register.form.firstName.placeholder"
                required=""
                value=""
              />
            </div>
            <div
              class="required field"
            >
              <label
                for="lastName"
              >
                register.form.lastName
              </label>
              <input
                data-testid="lastName"
                id="lastName"
                maxlength="255"
                name="lastName"
                placeholder="register.form.lastName.placeholder"
                required=""
                value=""
              />
            </div>
          </div>
          <div
            class="required field"
          >
            <label
              for="club"
            >
              register.form.club
            </label>
            <input
              data-testid="club"
              id="club"
              maxlength="255"
              name="club"
              placeholder="register.form.club.placeholder"
              required=""
              value=""
            />
          </div>
          <div
            class="required field"
          >
            <label
              for="email"
            >
              register.form.email
            </label>
            <input
              autocomplete="email"
              data-testid="email"
              id="email"
              maxlength="255"
              name="email"
              placeholder="register.form.email.placeholder"
              required=""
              type="email"
              value=""
            />
            <div
              class="ui error message"
              id="email-error"
            >
              <div
                class="content"
              >
                <p>
                  register.form.email.error.message
                </p>
              </div>
            </div>
          </div>
          <h3
            class="ui dividing header"
          >
            register.form.address
          </h3>
          <div
            class="sixteen fields"
          >
            <div
              class="required thirteen wide field"
            >
              <label
                for="street"
              >
                register.form.address.street
              </label>
              <input
                autocomplete="street-address"
                data-testid="street"
                id="street"
                maxlength="255"
                name="street"
                placeholder="register.form.address.street.placeholder"
                required=""
                value=""
              />
            </div>
            <div
              class="required three wide field"
            >
              <label
                for="no"
              >
                register.form.address.no
              </label>
              <input
                data-testid="no"
                id="no"
                maxlength="10"
                name="no"
                placeholder="register.form.address.no.placeholder"
                required=""
                value=""
              />
            </div>
          </div>
          <div
            class="equal width fields"
          >
            <div
              class="required field"
            >
              <label
                for="npa"
              >
                register.form.address.npa
              </label>
              <input
                autocomplete="postal-code"
                data-testid="npa"
                id="npa"
                maxlength="255"
                name="npa"
                placeholder="register.form.address.npa.placeholder"
                required=""
                type="number"
                value=""
              />
            </div>
            <div
              class="required field"
            >
              <label
                for="locality"
              >
                register.form.address.locality
              </label>
              <input
                data-testid="locality"
                id="locality"
                maxlength="255"
                name="locality"
                placeholder="register.form.address.locality.placeholder"
                required=""
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="ui segment"
        >
          <h2
            class="ui dividing header"
          >
            register.form.counts
          </h2>
          <p>
            register.form.counts.description
          </p>
          <h3
            class="ui dividing header"
          >
            register.form.food
          </h3>
          <p>
            register.form.food.description
          </p>
          <div
            class="equal width fields"
          >
            <div
              class="field QuantitySelector full-width"
            >
              <div
                class="ui segment QuantitySelector__Container"
              >
                <label
                  class="QuantitySelector__Label"
                  for="dinner"
                >
                  register.form.counts.dinner (CHF 45)
                </label>
                <div
                  aria-atomic="true"
                  aria-hidden="true"
                  aria-live="polite"
                  aria-relevant="all"
                  class="visually-hidden"
                >
                  0
                </div>
                <div
                  class="QuantitySelector__Controls"
                >
                  <button
                    class="ui icon disabled button QuantitySelector__Button QuantitySelector__Button--minus"
                    data-testid="dinner-minus"
                    disabled=""
                    tabindex="-1"
                    title="quantityselector.remove"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="minus icon"
                    />
                  </button>
                  <input
                    class="glow QuantitySelector__Quantity"
                    data-testid="dinner"
                    id="dinner"
                    max="100"
                    maxlength="3"
                    min="0"
                    minlength="1"
                    pattern="[0-9]*"
                    type="number"
                    value=""
                  />
                  <button
                    class="ui icon button QuantitySelector__Button QuantitySelector__Button--plus"
                    data-testid="dinner-plus"
                    title="quantityselector.add"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="plus icon"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              class="field QuantitySelector full-width"
            >
              <div
                class="ui segment QuantitySelector__Container"
              >
                <label
                  class="QuantitySelector__Label"
                  for="vegetarian"
                >
                  register.form.counts.vegetarian (CHF 45)
                </label>
                <div
                  aria-atomic="true"
                  aria-hidden="true"
                  aria-live="polite"
                  aria-relevant="all"
                  class="visually-hidden"
                >
                  0
                </div>
                <div
                  class="QuantitySelector__Controls"
                >
                  <button
                    class="ui icon disabled button QuantitySelector__Button QuantitySelector__Button--minus"
                    data-testid="vegetarian-minus"
                    disabled=""
                    tabindex="-1"
                    title="quantityselector.remove"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="minus icon"
                    />
                  </button>
                  <input
                    class="glow QuantitySelector__Quantity"
                    data-testid="vegetarian"
                    id="vegetarian"
                    max="100"
                    maxlength="3"
                    min="0"
                    minlength="1"
                    pattern="[0-9]*"
                    type="number"
                    value=""
                  />
                  <button
                    class="ui icon button QuantitySelector__Button QuantitySelector__Button--plus"
                    data-testid="vegetarian-plus"
                    title="quantityselector.add"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="plus icon"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="equal width fields"
          >
            <div
              class="field QuantitySelector full-width"
            >
              <div
                class="ui segment QuantitySelector__Container"
              >
                <label
                  class="QuantitySelector__Label"
                  for="dinnerKid"
                >
                  register.form.counts.dinnerKid (CHF 20)
                </label>
                <div
                  aria-atomic="true"
                  aria-hidden="true"
                  aria-live="polite"
                  aria-relevant="all"
                  class="visually-hidden"
                >
                  0
                </div>
                <div
                  class="QuantitySelector__Controls"
                >
                  <button
                    class="ui icon disabled button QuantitySelector__Button QuantitySelector__Button--minus"
                    data-testid="dinnerKid-minus"
                    disabled=""
                    tabindex="-1"
                    title="quantityselector.remove"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="minus icon"
                    />
                  </button>
                  <input
                    class="glow QuantitySelector__Quantity"
                    data-testid="dinnerKid"
                    id="dinnerKid"
                    max="100"
                    maxlength="3"
                    min="0"
                    minlength="1"
                    pattern="[0-9]*"
                    type="number"
                    value=""
                  />
                  <button
                    class="ui icon button QuantitySelector__Button QuantitySelector__Button--plus"
                    data-testid="dinnerKid-plus"
                    title="quantityselector.add"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="plus icon"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              class="field QuantitySelector full-width"
            >
              <div
                class="ui segment QuantitySelector__Container"
              >
                <label
                  class="QuantitySelector__Label"
                  for="vegetarianKid"
                >
                  register.form.counts.vegetarianKid (CHF 20)
                </label>
                <div
                  aria-atomic="true"
                  aria-hidden="true"
                  aria-live="polite"
                  aria-relevant="all"
                  class="visually-hidden"
                >
                  0
                </div>
                <div
                  class="QuantitySelector__Controls"
                >
                  <button
                    class="ui icon disabled button QuantitySelector__Button QuantitySelector__Button--minus"
                    data-testid="vegetarianKid-minus"
                    disabled=""
                    tabindex="-1"
                    title="quantityselector.remove"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="minus icon"
                    />
                  </button>
                  <input
                    class="glow QuantitySelector__Quantity"
                    data-testid="vegetarianKid"
                    id="vegetarianKid"
                    max="100"
                    maxlength="3"
                    min="0"
                    minlength="1"
                    pattern="[0-9]*"
                    type="number"
                    value=""
                  />
                  <button
                    class="ui icon button QuantitySelector__Button QuantitySelector__Button--plus"
                    data-testid="vegetarianKid-plus"
                    title="quantityselector.add"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="plus icon"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="equal width fields"
          >
            <div
              class="field QuantitySelector full-width"
            >
              <div
                class="ui segment QuantitySelector__Container"
              >
                <label
                  class="QuantitySelector__Label"
                  for="breakfast"
                >
                  register.form.counts.breakfast (CHF 10)
                </label>
                <div
                  aria-atomic="true"
                  aria-hidden="true"
                  aria-live="polite"
                  aria-relevant="all"
                  class="visually-hidden"
                >
                  0
                </div>
                <div
                  class="QuantitySelector__Controls"
                >
                  <button
                    class="ui icon disabled button QuantitySelector__Button QuantitySelector__Button--minus"
                    data-testid="breakfast-minus"
                    disabled=""
                    tabindex="-1"
                    title="quantityselector.remove"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="minus icon"
                    />
                  </button>
                  <input
                    class="glow QuantitySelector__Quantity"
                    data-testid="breakfast"
                    id="breakfast"
                    max="100"
                    maxlength="3"
                    min="0"
                    minlength="1"
                    pattern="[0-9]*"
                    type="number"
                    value=""
                  />
                  <button
                    class="ui icon button QuantitySelector__Button QuantitySelector__Button--plus"
                    data-testid="breakfast-plus"
                    title="quantityselector.add"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="plus icon"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              class="field QuantitySelector full-width"
            >
              <div
                class="ui segment QuantitySelector__Container"
              >
                <label
                  class="QuantitySelector__Label"
                  for="picknick"
                >
                  register.form.counts.picknick (noprice)
                </label>
                <div
                  aria-atomic="true"
                  aria-hidden="true"
                  aria-live="polite"
                  aria-relevant="all"
                  class="visually-hidden"
                >
                  0
                </div>
                <div
                  class="QuantitySelector__Controls"
                >
                  <button
                    class="ui icon disabled button QuantitySelector__Button QuantitySelector__Button--minus"
                    data-testid="picknick-minus"
                    disabled=""
                    tabindex="-1"
                    title="quantityselector.remove"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="minus icon"
                    />
                  </button>
                  <input
                    class="glow QuantitySelector__Quantity"
                    data-testid="picknick"
                    id="picknick"
                    max="100"
                    maxlength="3"
                    min="0"
                    minlength="1"
                    pattern="[0-9]*"
                    type="number"
                    value=""
                  />
                  <button
                    class="ui icon button QuantitySelector__Button QuantitySelector__Button--plus"
                    data-testid="picknick-plus"
                    title="quantityselector.add"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="plus icon"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
          <h3
            class="ui dividing header"
          >
            register.form.accommodation
          </h3>
          <p>
            register.form.accommodation.description
          </p>
          <div
            class="equal width fields"
          >
            <div
              class="field QuantitySelector full-width"
            >
              <div
                class="ui segment QuantitySelector__Container"
              >
                <label
                  class="QuantitySelector__Label"
                  for="sleepingAtGym"
                >
                  register.form.counts.sleepingAtGym (CHF 10)
                </label>
                <div
                  aria-atomic="true"
                  aria-hidden="true"
                  aria-live="polite"
                  aria-relevant="all"
                  class="visually-hidden"
                >
                  0
                </div>
                <div
                  class="QuantitySelector__Controls"
                >
                  <button
                    class="ui icon disabled button QuantitySelector__Button QuantitySelector__Button--minus"
                    data-testid="sleepingAtGym-minus"
                    disabled=""
                    tabindex="-1"
                    title="quantityselector.remove"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="minus icon"
                    />
                  </button>
                  <input
                    class="glow QuantitySelector__Quantity"
                    data-testid="sleepingAtGym"
                    id="sleepingAtGym"
                    max="100"
                    maxlength="3"
                    min="0"
                    minlength="1"
                    pattern="[0-9]*"
                    type="number"
                    value=""
                  />
                  <button
                    class="ui icon button QuantitySelector__Button QuantitySelector__Button--plus"
                    data-testid="sleepingAtGym-plus"
                    title="quantityselector.add"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="plus icon"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              class="field QuantitySelector full-width"
            >
              <div
                class="ui segment QuantitySelector__Container"
              >
                <label
                  class="QuantitySelector__Label"
                  for="camping"
                >
                  register.form.counts.camping (CHF 10)
                </label>
                <div
                  aria-atomic="true"
                  aria-hidden="true"
                  aria-live="polite"
                  aria-relevant="all"
                  class="visually-hidden"
                >
                  0
                </div>
                <div
                  class="QuantitySelector__Controls"
                >
                  <button
                    class="ui icon disabled button QuantitySelector__Button QuantitySelector__Button--minus"
                    data-testid="camping-minus"
                    disabled=""
                    tabindex="-1"
                    title="quantityselector.remove"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="minus icon"
                    />
                  </button>
                  <input
                    class="glow QuantitySelector__Quantity"
                    data-testid="camping"
                    id="camping"
                    max="100"
                    maxlength="3"
                    min="0"
                    minlength="1"
                    pattern="[0-9]*"
                    type="number"
                    value=""
                  />
                  <button
                    class="ui icon button QuantitySelector__Button QuantitySelector__Button--plus"
                    data-testid="camping-plus"
                    title="quantityselector.add"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="plus icon"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ui error message"
            data-testid="total-error"
          >
            <div
              class="content"
            >
              <p>
                register.form.error.total
              </p>
            </div>
          </div>
          <div
            class="RegisterForm__total-price"
          >
            register.form.counts.total
             : CHF 
            0
          </div>
        </div>
        <div
          class="ui error message"
          data-testid="validation-error"
        >
          <div
            class="content"
          >
            <p>
              register.form.validation.error
            </p>
          </div>
        </div>
        <div
          class="ui error message"
        >
          <div
            class="content"
          >
            <p>
              register.form.error.recaptcha
            </p>
          </div>
        </div>
        <div
          style="height: 78px; width: 302px;"
        >
           
        </div>
        <div
          class="RegisterForm__submit-button-container"
        >
          <button
            class="ui black button full-width RegisterForm__submit-button"
            type="submit"
          >
            register.form.submit
          </button>
        </div>
      </form>
    </div>
  </div>
</body>
`;
